provider/alibabacloud/alibaba_cloud.go

```diff
@@ -373,24 +374,24 @@ func (p *AlibabaCloudProvider) groupRecords(records []alidns.Record) (endpointMa
 func (p *AlibabaCloudProvider) records() ([]alidns.Record, error) {
 	log.Infof("Retrieving Alibaba Cloud DNS Domain Records")
 	var results []alidns.Record
-
+	hostedZoneNames, err := p.getDomainList()
+	if err != nil {
+		log.Errorf("AlibabaCloudProvider getDomainList error %v", err)
+		return results, err
+	}
 	if len(p.domainFilter.Filters) == 0 {
-		domainNames, tmpErr := p.getDomainList()
-		if tmpErr != nil {
-			log.Errorf("AlibabaCloudProvider getDomainList error %v", tmpErr)
-			return results, tmpErr
-		}
-		for _, tmpDomainName := range domainNames {
-			tmpResults, err := p.getDomainRecords(tmpDomainName)
+		var tmpResults []alidns.Record
+		for _, zone := range hostedZoneNames {
+			tmpResults, err = p.getDomainRecords(zone)
```

Comment:
```suggestion
			domainRecords, err := p.getDomainRecords(zone)
```

changing the append() statement below accordingly.